### PR TITLE
Feat: Migrate test setup to testing-library

### PR DESCRIPTION
**Pull request by irfan-maulana-tkp**
- Upgrade many dependencies
- Remove Enzyme
- Migrate to testing-library
- Add Github Action to run test in every push

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+module.exports = {
+	testMatch: [
+		"<rootDir>/src/**/__tests__/**/*.{js,jsx}",
+		"<rootDir>/src/**/*.{spec,test}.{js,jsx}"
+	],
+	transform: {
+		'^.+\\.(js|jsx|ts|tsx)$': '<rootDir>/node_modules/babel-jest',
```
Comment (jackyef) [inline jest.config.js]: Prefer using `ts-jest` for typescript files for more accurate coverage data.

https://github.com/facebook/jest/issues/7804

Comment (irfan-maulana-tkp) [reply jest.config.js]: Removing `ts` and `tsx` for now, since we are not support it yet

File: jest.config.js [hunk 2]
```diff
@@ -4,12 +4,12 @@ module.exports = {
 		"<rootDir>/src/**/*.{spec,test}.{js,jsx}"
 	],
 	transform: {
-		'^.+\\.(js|jsx|ts|tsx)$': '<rootDir>/node_modules/babel-jest',
+		'^.+\\.(js|jsx)$': '<rootDir>/node_modules/babel-jest',
 		'^.+\\.(css|less)$': '<rootDir>/jest-transformer/css.js',
-    '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '<rootDir>/jest-transformer/assets.js',
+    '^(?!.*\\.(js|jsxcss|json)$)': '<rootDir>/jest-transformer/assets.js',
```
Comment (jackyef) [inline jest.config.js]: typo keknya

Comment (irfan-maulana-tkp) [reply jest.config.js]: done

**Pull request merged**